zypper.py

```diff
@@ -0,0 +1,274 @@
+                info.info({field_str: '1'})
+
+
+def print_pending_updates(data, all_info, filters=None):
+    if all_info:
+        fields = [("Repository", "repository"),
+                  ("Name", "package-name"),
+                  ("Available Version",
+                   "available-version")]
+    else:
+        fields = [("Repository", "repository"),
+                  ("Name", "package-name")]
+    prefix = "zypper_update_pending"
+    description = (
+        "zypper package update available from repository. "
+        "(0 = not available, 1 = available)"
+    )
+    info = Info(prefix, description)
+
+    __print_pending_data(data, fields, info, filters)
+
+
+def print_pending_patches(data, all_info, filters=None):
+    if all_info:
+        fields = [("Repository", "repository"),
+                  ("Name", "patch-name"),
+                  ("Category", "category"),
+                  ("Severity", "severity"),
+                  ("Interactive", "interactive"),
+                  ("Status", "status")]
+    else:
+        fields = [("Repository", "repository"),
+                  ("Name", "patch-name"),
+                  ("Interactive", "interactive"),
+                  ("Status", "status")]
+    prefix = "zypper_patch_pending"
+    description = "zypper patch available from repository. (0 = not available , 1 = available)"
+    info = Info(prefix, description)
+
+    __print_pending_data(data, fields, info, filters)
+
+
+def print_orphaned_packages(data, filters=None):
+    fields = [("Name", "package"),
+              ("Version", "installed-version")]
+    prefix = "zypper_package_orphan"
+    description = "zypper packages with no update source (orphaned)"
+    info = Info(prefix, description)
+
+    __print_pending_data(data, fields, info, filters)
+
+
+def print_data_sum(data, prefix, description, filters=None):
+    gauge = Gauge(prefix,
+                  description,
+                  namespace=NAMESPACE,
+                  registry=REGISTRY)
+    filters = filters or {}
+    if len(data) == 0:
+        gauge.set(0)
+    else:
+        for package in data:
+            check = all(package.get(k) == v for k, v in filters.items())
+            if check:
+                gauge.inc()
+
+
+def print_reboot_required():
+    needs_restarting_path = '/usr/bin/needs-restarting'
+    is_path_ok = os.path.isfile(needs_restarting_path) and os.access(needs_restarting_path, os.X_OK)
+
+    if is_path_ok:
+        prefix = "node_reboot_required"
+        description = (
+            "Node require reboot to activate installed updates or patches. "
+            "(0 = not needed, 1 = needed)"
+        )
+        info = Info(prefix, description)
+        result = subprocess.run(
+            [needs_restarting_path, '-r'],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=False)
+        if result.returncode == 0:
+            info.info({"node_reboot_required": "0"})
```

Comment:
Once again you are abusing the `Info` metric type. Specifying a 0 or 1 value as a label will create new series whenever that value changes, leading to stale metrics due to Prometheus' look-behind mechanism. What you need to use is a _Gauge_ metric.